code/game/gamemodes/nuclear/pinpointer.dm

```diff
@@ -197,60 +242,43 @@
 //nuke op pinpointers//
 ///////////////////////
 /obj/item/pinpointer/nukeop
-	var/mode = 0	//Mode 0 locates disk, mode 1 locates the shuttle
 	var/obj/docking_port/mobile/home = null
 	slot_flags = SLOT_BELT | SLOT_PDA
+	syndicate = TRUE
+	modes = list(MODE_DISK, MODE_NUKE)
 
-/obj/item/pinpointer/nukeop/attack_self(mob/user as mob)
-	if(!active)
-		active = 1
-		if(!mode)
+/obj/item/pinpointer/nukeop/process()
+	switch(mode)
+		if(MODE_DISK)
 			workdisk()
-			to_chat(user, "<span class='notice'>Authentication Disk Locator active.</span>")
-		else
+		if(MODE_NUKE)
+			workbomb()
+		if(MODE_SHIP)
 			worklocation()
-			to_chat(user, "<span class='notice'>Shuttle Locator active.</span>")
-	else
-		active = 0
-		icon_state = icon_off
-		to_chat(user, "<span class='notice'>You deactivate the pinpointer.</span>")
 
 /obj/item/pinpointer/nukeop/workdisk()
-	if(!active) return
-	if(mode)		//Check in case the mode changes while operating
-		worklocation()
-		return
 	if(GLOB.bomb_set)	//If the bomb is set, lead to the shuttle
-		mode = 1	//Ensures worklocation() continues to work
+		mode = MODE_SHIP	//Ensures worklocation() continues to work
 		worklocation()
```

Comment:
Should not be needed since the next process tick also calls this

Reply:
Removed